react/features/mobile/callkit/CallKit.js

```diff
@@ -0,0 +1,235 @@
+ *  - performSetMutedCallAction: The call muted state should change. The
+ *    ancillary `data` object contains a `muted` attribute.
+ *  - providerDidReset: The system has reset, all calls should be terminated.
+ *    This event gets no associated data.
+ *
+ * All events get a `data` object with a `callUUID` property, unless stated
+ * otherwise.
+ */
+class CallKit extends NativeEventEmitter {
+    /**
+     * Initializes a new {@code CallKit} instance.
+     */
+    constructor() {
+        super(RNCallKit);
+        this._setup = false;
+    }
+
+    /**
+     * Returns True if the current platform is supported, false otherwise. The
+     * supported platforms are: iOS >= 10.
+     *
+     * @private
+     * @returns {boolean}
+     */
+    static isSupported() {
+        return Platform.OS === 'ios' && parseInt(Platform.Version, 10) >= 10;
+    }
+
+    /**
+     * Checks if CallKit was setup, and throws an exception in that case.
+     *
+     * @private
+     * @returns {void}
+     */
+    _checkSetup() {
+        if (!this._setup) {
+            throw new Error('CallKit not initialized, call setup() first.');
+        }
+    }
+
+    /**
+     * Adds a listener for the given event.
+     *
+     * @param {string} event - Name of the event we are interested in.
+     * @param {Function} listener - Function which will be called when the
+     * desired event is emitted.
+     * @returns {void}
+     */
+    addEventListener(event, listener) {
+        this._checkSetup();
+        if (!CallKit.isSupported()) {
+            return;
+        }
+
+        this.addListener(event, listener);
+    }
+
+    /**
+     * Notifies CallKit about an incoming call. This will display the system
+     * incoming call view.
+     *
+     * @param {string} uuid - Unique identifier for the call.
+     * @param {string} handle - Call handle in CallKit's terms. The room URL.
+     * @param {boolean} hasVideo - True if it's a video call, false otherwise.
+     * @returns {Promise}
+     */
+    displayIncomingCall(uuid, handle, hasVideo = true) {
+        this._checkSetup();
+        if (!CallKit.isSupported()) {
+            return Promise.resolve();
+        }
+
+        return RNCallKit.displayIncomingCall(uuid, handle, hasVideo);
+    }
+
+    /**
+     * Request CallKit to end the call.
+     *
+     * @param {string} uuid - Unique identifier for the call.
+     * @returns {Promise}
+     */
+    endCall(uuid) {
+        this._checkSetup();
+        if (!CallKit.isSupported()) {
+            return Promise.resolve();
+        }
+
+        return RNCallKit.endCall(uuid);
+    }
+
+    /**
+     * Removes a listener for the given event.
+     *
+     * @param {string} event - Name of the event we are no longer interested in.
+     * @param {Function} listener - Function which used to be called when the
+     * desired event was emitted.
+     * @returns {void}
+     */
+    removeEventListener(event, listener) {
```

Comment:
That's unnecessary not only because it introduces the confusing between `removeEventListener` and `removeListener` but also because `addListener` returns a subscription which may be `remove`d.